rudder-core/src/main/scala/com/normation/rudder/datasources/UpdateHttpDataset.scala

```diff
@@ -0,0 +1,277 @@
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.datasources
+
+import scala.collection.immutable.TreeMap
+import scala.concurrent.duration.Duration
+import scala.util.control.NonFatal
+
+import com.jayway.jsonpath.Configuration
+import com.jayway.jsonpath.DocumentContext
+import com.jayway.jsonpath.JsonPath
+import com.normation.cfclerk.domain.Variable
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.domain.nodes.NodeInfo
+import com.normation.rudder.domain.nodes.NodeProperty
+import com.normation.rudder.domain.parameters.Parameter
+import com.normation.rudder.domain.parameters.ParameterName
+import com.normation.rudder.services.policies.InterpolatedValueCompiler
+import com.normation.rudder.services.policies.InterpolationContext
+import com.normation.utils.Control._
+
+import net.liftweb.common.Box
+import net.liftweb.common.Empty
+import net.liftweb.common.Failure
+import net.liftweb.common.Full
+import net.liftweb.util.Helpers.tryo
+import net.minidev.json.JSONArray
+import net.minidev.json.JSONAware
+import net.minidev.json.JSONValue
+import scalaj.http.Http
+import scalaj.http.HttpOptions
+import scalaz._
+import scalaz.concurrent.Task
+
+/*
+ * This file contain the logic to update dataset from an
+ * HTTP Datasource.
+ * More specifically, it allows to:
+ * - query a node endpoint using node specific properties,
+ * - select a sub-json,
+ * - save it in the node properties.
+ */
+
+
+/*
+ * The whole service:
+ * - from a datasource and node context,
+ * - get the node datasource URL,
+ * - query it,
+ * - parse the json result,
+ * - return a rudder property with the content.
+ */
+class GetDataset(valueCompiler: InterpolatedValueCompiler) {
+
+  val compiler = new InterpolateNode(valueCompiler)
+
+  def getNode(datasourceName: DataSourceName, datasource: HttpDataSourceType, node: NodeInfo, policyServer: NodeInfo, parameters: Set[Parameter], connectionTimeout: Duration, readTimeOut: Duration): Box[NodeProperty] = {
+    for {
+      p          <- sequence(parameters.toSeq)(compiler.compileParameters) ?~! "Error when transforming Rudder Parameter for variable interpolation"
+      parameters =  p.toMap
+      url        <- compiler.compileInput(datasource.url, node, policyServer, parameters) ?~! s"Error when trying to parse URL ${datasource.url}"
+      path       <- compiler.compileInput(datasource.path, node, policyServer, parameters) ?~! s"Error when trying to compile JSON path ${datasource.path}"
+      time_0     =  System.currentTimeMillis
+      body       <- QueryHttp.GET(url, datasource.headers, datasource.sslCheck, connectionTimeout, readTimeOut) ?~! s"Error when fetching data from ${url}"
```

Comment:
We should handle get or post request depending on wether the datasource is configured for get or POST